provider/resources.go

```diff
@@ -156,6 +159,65 @@ func boolRef(b bool) *bool {
 	return &b
 }
 
+// stringValue gets a string value from a property map if present, else ""
+func stringValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []string) string {
+	val, ok := vars[prop]
+	if ok && val.IsString() {
+		return val.StringValue()
+	}
+	for _, env := range envs {
+		val, ok := os.LookupEnv(env)
+		if ok {
+			return val
+		}
+	}
+	return ""
+}
+
+func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) error {
+
+	// explicitly check to make sure that the user has a project available before we do
+	// anything with the provider
+	project := stringValue(vars, "project", []string{"GOOGLE_PROJECT", "GOOGLE_CLOUD_PROJECT",
```

Comment:
I'd suggest keeping each env var on a separate line to improve readability here.